### ISSUE TITLE
android 10: apps don't realize the network is up (compared to android 8.1)

### ISSUE THREAD

**Issue by bughit** [Genymobile/gnirehtet#223]
gnirehtet-rust-win64-v2.3.zip on win7

For example

- f-droid 1.7.1
- podcast addict 4.10.4
- Gmail 2019.08.04.263630132.release

On android 8.1 the above apps are able to connect when the gnirehtet vpn is active.

On android 10 they don't, report no connection.

Not all apps are affected (Firefox, Chrome work), but enough that gnirehtet is no longer usable.

This looks like an Android bug, but perhaps you can work around it, and also report it (you would know better what they screwed up)

**Comment by rom1v** [comment]
Do you get the confirmation dialog when you start?

Could you try the current `master` branch, please?
In particular, f8681809a28a0073f791f0d9eae65da02ae73c3d and cfc63a2c03bb10232ce04a687470d7594b4895f8 should help.

**Comment by bughit** [comment]
> Do you get the confirmation dialog when you start?

No. It starts and works, but a significant % apps don't realize the network is up and so fail to connect.

> Could you try the current `master` branch, please?

If you do a beta release, I am not set up for building.

**Comment by rom1v** [comment]
> If you do a beta release, I am not set up for building.

Here is a java version: [`gnirehtet-java-v2.3-19-g2a22b30.zip`](https://tmp.rom1v.com/gnirehtet/gnirehtet-java-v2.3-19-g2a22b30.zip)
_SHA256: 9245ff7f176a8d04338d1df0e7395b605f9b585903f52ce153fa959d12068002_

Execute `gnirehtet uninstall` first (because the version number has not been bumped, it won't update it automatically).

**Comment by bughit** [comment]
no change, same problem

**Comment by rom1v** [comment]
OK. Thank you for the test.

(you got the [confirmation dialog](https://raw.githubusercontent.com/Genymobile/gnirehtet/master/assets/request.jpg) now, I guess, after you uninstalled)

**Comment by bughit** [comment]
yes

**Comment by bughit** [comment]
Is it worth trying 2.4?

**Comment by bughit** [comment]
no change in 2.4

**Comment by thelazaf** [comment]
I'm getting an error called

 Error Type 3
Activity Class {com.genymobile.gnirehtet/com.genymobile.gnirehtertgnirehtetactivity} does not exist.

Does anybody know how to fix this? I'm using OnePlus 6 with Android 10 and it isn't rooted

**Comment by rom1v** [comment]
You're probably using an old APK with a new gnirehtet binary.

Make sure you use the right one, and execute `gnirehtet uninstall` to uninstall any previous version on your device.

**Comment by thelazaf** [comment]
It was working fine day before, and I don't use the commands, I just plug in my device and run the gnirehtet-run file. This error started all of a sudden out of nowhere

**Comment by thelazaf** [comment]
Tried uninstalling using command and ran it again, working now. Thanks!

**Comment by bughit** [comment]
@rom1v 

Decided to try it with android 11 and v2.5, the problem remains. Do you only maintain it for older android versions?

I am curious where the fault lies. Does android provide an API for checking network availability (as opposed to just wifi)? Does it work with gnirehtet? Do some apps (mentioned here) not use it?
